src/base/bittorrent/session.cpp

```diff
@@ -4480,54 +4480,63 @@ void Session::startUpTorrents()
-                            resumeData.ltAddTorrentParams.ti = altResumeData.ltAddTorrentParams.ti;
-                        }
+                        LoadTorrentParams altResumeData = *loadAltResumeDataResult;
+                        resumeData.ltAddTorrentParams.ti = altResumeData.ltAddTorrentParams.ti;
                     }
-
-                    // remove alternative "resume data" and skip the attempt to load it
-                    m_resumeDataStorage->remove(torrentIDv1);
-                    skippedIDs.insert(torrentIDv1);
                 }
+
+                // remove alternative "resume data" and skip the attempt to load it
+                m_resumeDataStorage->remove(torrentIDv1);
+                skippedIDs.insert(torrentIDv1);
             }
-            else
+        }
+        else if (torrentID == torrentIDv1)
+        {
+            torrentID = torrentIDv2;
+            needStore = true;
+            m_resumeDataStorage->remove(torrentIDv1);
+
+            if (indexedTorrents.contains(torrentID))
             {
-                torrentID = torrentIDv2;
-                needStore = true;
-                m_resumeDataStorage->remove(torrentIDv1);
+                skippedIDs.insert(torrentID);
 
-                if (indexedTorrents.contains(torrentID))
+                const std::optional<LoadTorrentParams> loadPreferredResumeDataResult = startupStorage->load(torrentID);
+                if (loadPreferredResumeDataResult)
                 {
-                    skippedIDs.insert(torrentID);
-
-                    const std::optional<LoadTorrentParams> loadPreferredResumeDataResult = startupStorage->load(torrentID);
-                    if (loadPreferredResumeDataResult)
-                    {
-                        LoadTorrentParams preferredResumeData = *loadPreferredResumeDataResult;
-                        std::shared_ptr<lt::torrent_info> ti = resumeData.ltAddTorrentParams.ti;
-                        if (!preferredResumeData.ltAddTorrentParams.ti)
-                            preferredResumeData.ltAddTorrentParams.ti = ti;
+                    LoadTorrentParams preferredResumeData = *loadPreferredResumeDataResult;
+                    std::shared_ptr<lt::torrent_info> ti = resumeData.ltAddTorrentParams.ti;
+                    if (!preferredResumeData.ltAddTorrentParams.ti)
+                        preferredResumeData.ltAddTorrentParams.ti = ti;
 
-                        resumeData = preferredResumeData;
-                    }
+                    resumeData = preferredResumeData;
                 }
             }
         }
+        else
+#else
+        const lt::sha1_hash infoHash = (resumeData.ltAddTorrentParams.ti
+                                          ? resumeData.ltAddTorrentParams.ti->info_hash()
+                                          : resumeData.ltAddTorrentParams.info_hash);
+        if (torrentID != TorrentID::fromInfoHash(infoHash))
 #endif
+        {
```

Comment:
Please don't split it over `#if`

Reply:
Then I will have to repeat some code in each branch. Is that okay with you?

Reply:
yes.